DDCameraViewController/Classes/DDScannerViewController.m

```diff
@@ -0,0 +1,52 @@
+//
+//  DDScannerViewController.m
+//  DDCameraViewController
+//
+//  Copyright (c) 2020 dashdevs.com. All rights reserved.
+//
+
+#import "DDScannerViewController.h"
+#import "DDCameraViewController_Private.h"
+
+@interface DDScannerViewController ()<AVCaptureMetadataOutputObjectsDelegate>
+@property (strong, nonatomic, nullable) NSString *machineReadableCode;
+@property (strong, nonatomic, nullable) NSArray<AVMetadataObjectType> *metadataObjectTypes;
+@property (strong, nonatomic, nullable) AVCaptureMetadataOutput *output;
+@end
+
+@implementation DDScannerViewController
+
+- (void)initializeCaptureSession {
+    [super initializeCaptureSession];
+    _output = [[AVCaptureMetadataOutput alloc] init];
+    [self.captureSession addOutput:_output];
+    [_output setMetadataObjectsDelegate:self queue:dd_capture_session_queue()];
```

Comment:
лучше использовать property вместо ivar: `[self.output ....]`. nonatomic не будет применен если мы работаем с ivar. ну или если нам не нужны геттеры и сеттеры, то можно просто объявить ivar:
```
@implementation DDScannerViewController {
   AVCaptureMetadataOutput *output;
}
```
но мне кажется лучше использовать проперти

Reply:
@yakovshkolnikov там вроде метод вызывается в init. использовать мутаторы в init считалось не очень хорошей практикой, чтобы мутатор не повлиял на инициализацию

Reply:
хотя при этом инициализация конечно должна быть непосредственно в init

Reply:
> When using properties, instance variables should always be accessed and mutated using self.. This means that all properties will be visually distinct, as they will all be prefaced with self..
> 
> An exception to this: inside initializers, the backing instance variable (i.e. _variableName) should be used directly to avoid any potential side effects of the getters/setters.

https://github.com/raywenderlich/objective-c-style-guide

> Avoid Messaging the Current Object Within Initializers and -dealloc

https://google.github.io/styleguide/objcguide.html